### DISCUSSION TITLE
Implementing imx8MN board.

### DISCUSSION THREAD

**Discussion by JaagupAverin** [zephyrproject-rtos/zephyr#49061] [Q&A]
Hello.

I am trying to build a Zephyr application for the imx8MN devkit. A board for imx8MM devkit has already [been implemented](https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm64/imx8mm_evk/doc/index.rst). However I am having trouble getting the Nano version of the board to work.
I have duplicated the existing board folder, but some things are unclear to me, such as what the sram0 value should be pointing to, and where it came from from in the [existing imx8MM code](https://github.com/zephyrproject-rtos/zephyr/blob/6fc4bb9486248b2d2a9cc8eff505f365a4006ef1/boards/arm64/imx8mm_evk/imx8mm_evk.dts#L27).

Furthermore, a dts has been implemented for the imx8MM board as [seen here](https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/arm64/nxp/nxp_imx8mm_a53.dtsi), however this implementation also confuses me, as it specifies 2 cores, whereas the actual devkit should have 4 cores.

Finally, what else should be changed (in terms of registers and sizes) when converting the existing Mini board to a Nano board? I have many gaps in understanding when it comes to devicetree and board implementation so any answers and general direction is very welcome!

Thank you.

**Comment by JiafeiPan** [comment]
For the sram0 address, you can use the same value with IMX8MM "0x93c00000", it is ram address used by Zephyr and specified by dts or CONFIG_SRAM_BASE_ADDRESS and CONFIG_SRAM_SIZE. If you boot Zephyr from uboot, you should use the same memory address to load and run Zephyr. If you use Jailhouse, the address should be same with the memory address in inmate cell configuration.

For the number of CPU Cores, current code in Zephyr community doesn't enable SMP, so only Core0 is used.

Finally, for IMX8MN platform support in Zephyr, NXP will upstream it to community soon, and it will enable SMP.

**Reply by JaagupAverin** [reply]
Thank you for the details.

I have just one more thing I would like to verify about the existing Mini board. It uses [nxp,imx-iuart](https://github.com/zephyrproject-rtos/zephyr/blob/84f3a0a8c6a930f0252d9ceca6adf8c3b1284af0/dts/arm64/nxp/nxp_imx8mm_a53.dtsi#L84) for its UART drivers, which is related to the [UART_IMX config](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/serial/Kconfig.imx). But that appears to support just the iMX7 family; and indeed, when I enable uart2 from the overlay and the UART_IMX config, I get missing header errors for [soc.h and uart_imx.h](https://github.com/zephyrproject-rtos/zephyr/blob/84f3a0a8c6a930f0252d9ceca6adf8c3b1284af0/drivers/serial/uart_imx.c#L19).

Does the IMX8MM board in fact have UART support at this time? If so, can you suggest what configuration might be needed to include the <soc.h> and <uart_imx.h> files properly?

**Reply by JiafeiPan** [reply]
Except soc.h, there are also some other dependencies, such as pinctrl and clock, I will upstream new platform support for imx8mm/p/n around the end of this month, hope it will be okay for you and feel free to let me know if you have more comments, thanks.

**Reply by JiafeiPan** [reply]
FYI, I have upstreamed new platfomr imx8mn, and also updated imx8mm and imxmp, please refer to the following PR and it is in review progress:

https://github.com/zephyrproject-rtos/zephyr/pull/49651
